src/components/File/File.jsx

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import classNames from "classnames";
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import * as Rivet from "../util/Rivet";
+import Icon from "../util/RivetIcons";
+
+const classPrefix = "rvt-file";
+
+function useForceUpdate() {
+  const [value, setValue] = React.useState(0); // integer state
+  return () => setValue((value) => value + 1); // update state to force render
+  // A function that increment 👆🏻 the previous state like here
+  // is better than directly setting `setValue(value + 1)`
+}
+
+const handleFileChange = (onChange, forceUpdate) => (changeEvent) => {
+  if (onChange) {
+    onChange(changeEvent);
+  }
+  forceUpdate();
+};
+
+const File = ({
+  className,
+  id = Rivet.shortuid(),
+  innerRef,
+  label,
+  multiple,
+  onChange,
+  secondary,
+  ...attrs
+}) => {
+  const forceUpdate = useForceUpdate();
+
+  const fileInput = innerRef || React.useRef();
```

Comment:
Hooks should never be conditional. I'm pretty sure the way of doing what you want is to pass the initial value to `useRef`

```suggestion
  const fileInput = React.useRef(innerRef);
```

Reply:
Fixing. Thanks.